### ISSUE TITLE
Configuration via environment variable substitution doesn't work anymore in Keycloak 21

### ISSUE THREAD

**Issue by deleze** [keycloak/keycloak#20609]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

dist/quarkus

### Describe the bug

The configuration via environment variable doesn't work anymore as describe in the documentation: https://www.keycloak.org/server/configuration#_using_environment_variables_for_configuration_values

In the version 21.1.1, the variable substitution in `keycloak.conf` doesn't work anymore. The same configuration was working with 20.0.3.

- `${KEYCLOAK_LOGLEVEL:INFO}` result as `INFO}` with a `}` at the end
- `${KEYCLOAK_LOGLEVEL}` result as `${KEYCLOAK_LOGLEVEL}`

### Version

21.1.1

### Expected behavior

Environment variable substitution : 

- `${KEYCLOAK_LOGLEVEL:INFO}` result as `INFO`
- `${KEYCLOAK_LOGLEVEL}` result as the value of the environment variable

### Actual behavior

Environment variable substitution : 

- `${KEYCLOAK_LOGLEVEL:INFO}` result as `INFO}`
- `${KEYCLOAK_LOGLEVEL}` result as `${KEYCLOAK_LOGLEVEL}`

### How to Reproduce?

Sample repository : https://github.com/deleze/keycloak-config-issue 

Content of `conf/keycloak.conf` with an environment variable substitution

```
log-console-color=true 
log-level=${KEYCLOAK_LOGLEVEL:INFO}
```

## 20.0.3 -> OK

```shell
docker build -t deleze/keycloak:20.0.3 -f ./Dockerfile_20.0.3 .
docker run --rm --name k20 -p 8080:8080 deleze/keycloak:20.0.3
```
Open http://localhost:8080/auth/admin

## 21.1.1 -> NOT OK

```shell
docker build -t deleze/keycloak:21.1.1  .
docker run --rm --name k21 -p 8080:8080 deleze/keycloak:21.1.1
```

The server didn't start because the substitutions result of `${KEYCLOAK_LOGLEVEL:INFO}` result as `INFO}` with a `}` at the end.  

```shell
docker build -t deleze/keycloak:21.1.1  .
docker run --rm --name k21 -p 8080:8080 -e "KEYCLOAK_LOGLEVEL=DEBUG" deleze/keycloak:21.1.1
```
The server didn't start either.

**Comment by deleze** [comment]
There is a unit test to verify exactly this case... weird... https://github.com/keycloak/keycloak/blob/cffb8141e2d5748d5e99d5b0cd31cb1478927166/quarkus/runtime/src/test/java/org/keycloak/quarkus/runtime/configuration/test/ConfigurationTest.java#L131

https://github.com/keycloak/keycloak/blob/cffb8141e2d5748d5e99d5b0cd31cb1478927166/quarkus/runtime/src/test/resources/META-INF/keycloak.conf#L3

**Comment by vmuzikar** [comment]
Duplicate of https://github.com/keycloak/keycloak/issues/20032

@deleze Hello, thank you for the report. I believe this is a duplicate issue that has already been resolved in `main`. Could you please try it with the latest nightly release? I'm going to close this. Feel free to reopen if still valid.

**Comment by vmuzikar** [comment]
Duplicate of https://github.com/keycloak/keycloak/issues/20032

**Comment by deleze** [comment]
Hello I try with the quay.io/keycloak/keycloak:nightly image and it works!  Thank you.

@vmuzikar Did you know when will be the next release ?

**Comment by vmuzikar** [comment]
It should be included in 21.1.2 micro release but I can't share any ETA at the moment.

**Comment by deleze** [comment]
@vmuzikar I saw that the milestone 21.1.2 have no more open issue. Did you plan to release it ?
